plugins/applications/src/lib.rs

```diff
@@ -1,15 +1,30 @@
+use std::{env, fs, process::Command};
+
 use abi_stable::std_types::{ROption, RString, RVec};
-use anyrun_plugin::{anyrun_interface::HandleResult, *};
 use fuzzy_matcher::FuzzyMatcher;
-use scrubber::DesktopEntry;
 use serde::Deserialize;
-use std::{env, fs, process::Command};
+
+use anyrun_plugin::{*, anyrun_interface::HandleResult};
+use scrubber::DesktopEntry;
+
+use crate::execution_stats::ExecutionStats;
 
 #[derive(Deserialize)]
 pub struct Config {
-    desktop_actions: bool,
+    /// Limit amount of entries shown by the applications plugin (default: 5)
     max_entries: usize,
+    /// Whether to evaluate desktop actions as well as desktop applications (default: false)
+    desktop_actions: bool,
+    /// Whether to use a specific terminal or just the first terminal available (default: None)
     terminal: Option<String>,
+    /// Whether to put more often used applications higher in the search rankings (default: true)
+    use_usage_statistics: bool,
+    /// How much score to add for every usage of an application (default: 50)
+    /// Each matching letter is 25 points
+    usage_score_multiplier: i64,
```

Comment:
Since using a logarithm would disassociate the value from a specific ranking score, we should probably map this to something like:

```
0: 0% weighting - No effect
1: 100% weighting - Default
1.5: 150% weighting - Preference for weighted entries
```

This also makes the `use_usage_statistics` redundant. Though I like verbosity, so maybe we should keep it.

Reply:
I would definitely keep `use_usage_statistics` to enable users to keep filling the usage-file without doing anything with it.
Knowing developers, I'm sure someone will discover this feature and use it this way.

As mentioned in https://github.com/anyrun-org/anyrun/pull/176#discussion_r1687038693 I like the idea of logarithms, but the `usage_score_multiplier` and `max_counted_usages` still needs to exist to balance the score with the score from the fuzzymatcher from the search-results.
If these values (or a version of them) are not exposed, the recency score could overwhelm the actual matches to the query, thus always showing the most used entries.
In reverse, it could also not have any effect at all, if the multiplier is too low.